probe-rs/src/core/mod.rs

```diff
@@ -52,6 +52,75 @@ pub struct CoreInformation {
     pub pc: u32,
 }
 
+#[derive(Debug, Clone)]
+pub struct RegisterDescription {
+    pub(crate) name: &'static str,
+    pub(crate) kind: RegisterKind,
+    pub(crate) address: CoreRegisterAddress,
+}
+
+impl RegisterDescription {
+    pub fn name(&self) -> &'static str {
+        self.name
+    }
+}
+
+impl From<&RegisterDescription> for CoreRegisterAddress {
+    fn from(description: &RegisterDescription) -> CoreRegisterAddress {
+        description.address
+    }
+}
+
+#[derive(Debug, Clone, Copy)]
+pub(crate) enum RegisterKind {
+    General,
+    PC,
+}
+
+#[derive(Clone)]
+pub struct RegisterFile {
+    pub(crate) platform_registers: Vec<RegisterDescription>,
+
+    pub(crate) program_counter: RegisterDescription,
+
+    pub(crate) stack_pointer: RegisterDescription,
+
+    pub(crate) return_address: RegisterDescription,
+
+    pub(crate) argument_registers: Vec<RegisterDescription>,
+    pub(crate) result_registers: Vec<RegisterDescription>,
+}
+
+impl RegisterFile {
+    pub fn registers(&self) -> impl Iterator<Item = &RegisterDescription> {
+        self.platform_registers.iter()
+    }
+
+    pub fn program_counter(&self) -> &RegisterDescription {
+        &self.program_counter
+    }
+
+    pub fn stack_pointer(&self) -> &RegisterDescription {
+        &self.stack_pointer
+    }
+
+    pub fn return_address(&self) -> &RegisterDescription {
+        &self.return_address
+    }
+
+    pub fn argument_register(&self, index: usize) -> &RegisterDescription {
+        &self.argument_registers[index]
```

Comment:
This can panic internally. Not sure if I like this.

Reply:
I've added non-panicking versions of these functions.